subprojects/core/src/main/java/org/gradle/caching/internal/controller/impl/LifecycleAwareBuildCacheController.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.controller.impl;
+
+import org.gradle.caching.configuration.internal.BuildCacheConfigurationInternal;
+import org.gradle.caching.internal.controller.BuildCacheController;
+import org.gradle.internal.service.scopes.Scope;
+import org.gradle.internal.service.scopes.ServiceScope;
+
+@ServiceScope(Scope.Gradle.class)
+public interface LifecycleAwareBuildCacheController extends BuildCacheController {
```

Comment:
This is in core, alongside `RootBuildCacheControllerRef`, so as to keep the complexity of the BCC lifecycle in Gradle in one place.

The plan is that at some point we better model services who need data from the Gradle model, and this complexity can go away.

Reply:
I think it would be good to mention this in a comment in the code, too, perhaps as a `TODO`.

Reply:
Added